### PR TITLE
Enforce IDE0005 (remove unnecessary usings) as build warning in few IDE layers

### PR DESCRIPTION
**Pull request by mavasani**
_Recommended to review commit by commit_

Enables "Remove unnecessary usings" as build warning in following IDE layers: CodeStyle, Features and Workspaces. The IDE team plans to dogfood build enforcement of few code style rules which have recently been added to CodeStyle NuGet package

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/DocumentBasedFixAllProvider.cs [hunk 1]
```diff
@@ -9,7 +9,10 @@
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.CodeActions;
 using Microsoft.CodeAnalysis.PooledObjects;
+
+#if NETSTANDARD2_0 || CODE_STYLE
```
Comment (sharwell) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/DocumentBasedFixAllProvider.cs:13]: The code style layer is netstandard2.0, so this condition can be simplified.

```suggestion
#if NETSTANDARD2_0
```

File: src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs [hunk 2]
```diff
@@ -10,12 +10,14 @@
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.IO;
-using System.Linq;
```
Comment (mavasani) [inline src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs:13]: @dotnet/roslyn-compiler There are trivial changes to only a handful for compiler files here - these files are linked into both Compilers and Workspaces, hence cannot be avoided in this PR.

Comment (mavasani) [reply src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs:13]: @dotnet/roslyn-compiler - can someone please take a quick look? Change just removes unused usings in few files.

Comment (jcouv) [reply src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs:13]: Looking

---
In reply to: [405789406](https://github.com/dotnet/roslyn/pull/43166#discussion_r405789406) [](ancestors = 405789406)

Comment (jcouv) [reply src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs:13]: Signed off.
@mavasani Is the plan to enable this enforcement on the entire roslyn repo down the line?

Comment (mavasani) [reply src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs:13]: > Is the plan to enable this enforcement on the entire roslyn repo down the line?

Given these are code style rules, we understand that each team might have different preferences on the settings as well as whether or not to enforce a specific rule. We are planning to dogfood a few code style rules in IDE layer for couple of weeks, at which point we will decide which rules to retain in CI, and which ones to just keep for IDE experience. We can then follow-up with the compiler team to see if they are interested in enforcement of any such rules in CI, and if so, we can work towards enabling them in the entire repo.

**Pull request merged**